### PR TITLE
Add Command History DG 

### PR DESCRIPTION
**Pull request by wr1159**
Edited Developer Guide to add
* Proposed Command History implementation details
* Proposed Command History user stories 
Feedback required:
- Should I include Class Diagram
- Should I include Sequence Diagram (Not so sure how to do it for my case)
- Is the activity diagram not very useful?
- Is it okay for me to stack 2 class diagrams in one diagram?

### CODE REVIEW

**Review by dylkaw — APPROVED** [review]
LGTM! I think the activity diagram is fine.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -237,6 +237,70 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+![CommandHistoryState0](images/CommandHistoryState0.png)
+
+Step 2. The user presses '↑' while the commandBox is selected. `CommandHistory#getPreviousCommand()` is called and the previous command is displayed in the commandBox. 
+When the previous command is entered into the commandBox, the newly edited version of command will not be stored in `CommandHistory`.
+
+![CommandHistoryState1](images/CommandHistoryState1.png)
+
+Step 3. The user presses '↑' while the commandBox is selected. `CommandHistory#getPreviousCommand()` is called 
+and the previous command is displayed in the commandBox. When the previous command is edited and is entered into the commandBox, the newly edited version of command will be stored in `CommandHistory` after calling `CommandHistory#addCommand`.
```
Comment (SelwynAng) [inline docs/DeveloperGuide.md]: `CommandHistory#addCommand` should be `CommandHistory#addCommand()`.

**Review by SelwynAng — COMMENTED** [review]
LGTM aside from a small nit.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (wr1159) [reply docs/DeveloperGuide.md]: Thanks I have fixed this in https://github.com/AY2324S1-CS2103T-F11-1/tp/pull/163/commits/0777b776c9cd2502fa380f09ee4e11f0d1450543

**Pull request merged**